internal/dataplane/kongstate/route.go

```diff
@@ -218,18 +232,21 @@ func (r *Route) overrideSNIs(log logrus.FieldLogger, anns map[string]string) {
 
 // overrideByAnnotation sets Route protocols via annotation.
 func (r *Route) overrideByAnnotation(log logrus.FieldLogger) {
-	r.overrideProtocols(r.Ingress.Annotations)
 	r.overrideStripPath(r.Ingress.Annotations)
 	r.overrideHTTPSRedirectCode(r.Ingress.Annotations)
 	r.overridePreserveHost(r.Ingress.Annotations)
-	r.overrideRegexPriority(r.Ingress.Annotations)
-	r.overrideMethods(log, r.Ingress.Annotations)
-	r.overrideSNIs(log, r.Ingress.Annotations)
 	r.overrideRequestBuffering(log, r.Ingress.Annotations)
 	r.overrideResponseBuffering(log, r.Ingress.Annotations)
-	r.overrideHosts(log, r.Ingress.Annotations)
-	r.overrideHeaders(r.Ingress.Annotations)
-	r.overridePathHandling(log, r.Ingress.Annotations)
+
+	if !r.ExpressionRoutes {
```

Comment:
Maybe add a comment why do we need to skip those overrides ? It's not immediately obvious to me

Reply:
added comments here.